File: RestAPIs/Admin_Services/Models/Rating.cs
using Admin_Services.Models;

public class Rating
{
    public int RatingId { get; set; }

    public int FromUserId { get; set; }
    public User FromUser { get; set; }   // 🔥 REQUIRED

    public int ToUserId { get; set; }
    public User ToUser { get; set; }     // 🔥 REQUIRED

    public int ServiceId { get; set; }
    public Service Service { get; set; } // 🔥 REQUIRED

    public int Rating1 { get; set; }
    public string Review { get; set; }
    public DateTime? RatingDate { get; set; }
}

File: User Services/freelancer/src/main/java/com/worksphere/freelancer/entity/Rating.java
package com.worksphere.freelancer.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "rating")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Rating {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "rating_id")
    private Integer ratingId;

    @ManyToOne
    @JoinColumn(name = "from_user_id")
    private User fromUser;

    @ManyToOne
    @JoinColumn(name = "to_user_id")
    private User toUser;

    @ManyToOne
    @JoinColumn(name = "service_id")
    private Services service;

    @Column(name = "rating")
    private Integer rating;

    @Column(name = "review")
    private String review;

    @Column(name = "rating_date")
    private LocalDate ratingDate;
}
